File: sdk/v1.0.21/javascript/custom_albums.js
function stripUndef(o) { const r = {}; for (const k in o) if (o[k] !== undefined) r[k] = o[k]; return r; }

export class CustomAlbums {
  constructor(ctx) { this._ctx = ctx; }

  async getCustomAlbumDetailById(customAlbumId) {
    this._ctx.requireTokens();
    const path = `/api/v1/custom_album/${encodeURIComponent(customAlbumId)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getAllProjectCustomAlbums(projectTableName) {
    this._ctx.requireTokens();
    const path = `/api/v1/custom_album/project/${encodeURIComponent(projectTableName)}`;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getCustomAlbumPhotosById(customAlbumId, { ascOrDesc, lastId, limit } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/custom_album/photos/${encodeURIComponent(customAlbumId)}/`;
    const query = new URLSearchParams();
    if (ascOrDesc !== undefined) (Array.isArray(ascOrDesc) ? ascOrDesc : [ascOrDesc]).forEach(v => query.append('asc_or_desc', v));
    if (lastId !== undefined) (Array.isArray(lastId) ? lastId : [lastId]).forEach(v => query.append('last_id', v));
    if (limit !== undefined) (Array.isArray(limit) ? limit : [limit]).forEach(v => query.append('limit', v));
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async getRankedCustomAlbumById(customAlbumId, { ascOrDesc, lastId, limit } = {}) {
    this._ctx.requireTokens();
    let path = `/api/v1/custom_album/photos/ranked/${encodeURIComponent(customAlbumId)}/`;
    const query = new URLSearchParams();
    if (ascOrDesc !== undefined) (Array.isArray(ascOrDesc) ? ascOrDesc : [ascOrDesc]).forEach(v => query.append('asc_or_desc', v));
    if (lastId !== undefined) (Array.isArray(lastId) ? lastId : [lastId]).forEach(v => query.append('last_id', v));
    if (limit !== undefined) (Array.isArray(limit) ? limit : [limit]).forEach(v => query.append('limit', v));
    const qs = query.toString();
    if (qs) path += '?' + qs;
    const { data } = await this._ctx.client.request(path, 'GET', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }

  async createProjectCustomAlbum(projectTableName, name = undefined, description = undefined, photoIdInclusionList = undefined, photoIdRemovalList = undefined) {
    this._ctx.requireTokens();
    const path = `/api/v1/custom_album/project/${encodeURIComponent(projectTableName)}`;
    const body = stripUndef({
      name: name,
      description: description,
      photo_id_inclusion_list: photoIdInclusionList,
      photo_id_removal_list: photoIdRemovalList,
    });
    const { data } = await this._ctx.client.request(path, 'POST', this._ctx.accessToken, this._ctx.refreshToken, body);
    return data;
  }

  async updateCustomAlbum(albumId, name = undefined, description = undefined, photoIdInclusionList = undefined, photoIdRemovalList = undefined) {
    this._ctx.requireTokens();
    const path = `/api/v1/custom_album/${encodeURIComponent(albumId)}`;
    const body = stripUndef({
      name: name,
      description: description,
      photo_id_inclusion_list: photoIdInclusionList,
      photo_id_removal_list: photoIdRemovalList,
    });
    const { data } = await this._ctx.client.request(path, 'PUT', this._ctx.accessToken, this._ctx.refreshToken, body);
    return data;
  }

  async deleteCustomAlbum(albumId) {
    this._ctx.requireTokens();
    const path = `/api/v1/custom_album/${encodeURIComponent(albumId)}`;
    const { data } = await this._ctx.client.request(path, 'DELETE', this._ctx.accessToken, this._ctx.refreshToken);
    return data;
  }
}

File: sdk/v1.8.0-dev.92/python/mediaviz_sdk/custom_albums.py
from __future__ import annotations
from typing import Any
from urllib.parse import quote, urlencode
import httpx

from .errors import handle_response


class CustomAlbums:
    def __init__(self, ctx) -> None:
        self._ctx = ctx

    def get_custom_album_detail_by_id(self, custom_album_id: int) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/' + quote(str(custom_album_id), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_all_project_custom_albums(self, project_table_name: str) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/project/' + quote(str(project_table_name), safe='')
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_custom_album_photos_by_id(
        self,
        custom_album_id: int,
        asc_or_desc: str | None = None,
        last_id: Any | None = None,
        limit: Any | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/photos/' + quote(str(custom_album_id), safe='') + '/'
        _q: dict[str, Any] = {}
        if asc_or_desc is not None:
            _q['asc_or_desc'] = asc_or_desc
        if last_id is not None:
            _q['last_id'] = last_id
        if limit is not None:
            _q['limit'] = limit
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def get_ranked_custom_album_by_id(
        self,
        custom_album_id: int,
        asc_or_desc: str | None = None,
        last_id: Any | None = None,
        limit: Any | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/photos/ranked/' + quote(str(custom_album_id), safe='') + '/'
        _q: dict[str, Any] = {}
        if asc_or_desc is not None:
            _q['asc_or_desc'] = asc_or_desc
        if last_id is not None:
            _q['last_id'] = last_id
        if limit is not None:
            _q['limit'] = limit
        if _q:
            path += '?' + urlencode(_q, doseq=True)
        return self._ctx.client.request(path, 'GET', self._ctx.access_token, self._ctx.refresh_token).data

    def create_project_custom_album(
        self,
        project_table_name: str,
        *,
        name: str | None = None,
        description: str | None = None,
        photoIdInclusionList: list | None = None,
        photoIdRemovalList: list | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/project/' + quote(str(project_table_name), safe='')
        body = {k: v for k, v in {
            'name': name,
            'description': description,
            'photo_id_inclusion_list': photoIdInclusionList,
            'photo_id_removal_list': photoIdRemovalList,
        }.items() if v is not None}
        return self._ctx.client.request(path, 'POST', self._ctx.access_token, self._ctx.refresh_token, body).data

    def update_custom_album(
        self,
        album_id: int,
        *,
        name: str | None = None,
        description: str | None = None,
        photoIdInclusionList: list | None = None,
        photoIdRemovalList: list | None = None,
    ) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/' + quote(str(album_id), safe='')
        body = {k: v for k, v in {
            'name': name,
            'description': description,
            'photo_id_inclusion_list': photoIdInclusionList,
            'photo_id_removal_list': photoIdRemovalList,
        }.items() if v is not None}
        return self._ctx.client.request(path, 'PUT', self._ctx.access_token, self._ctx.refresh_token, body).data

    def delete_custom_album(self, album_id: int) -> dict[str, Any]:
        self._ctx.require_tokens()
        path = '/api/v1/custom_album/' + quote(str(album_id), safe='')
        return self._ctx.client.request(path, 'DELETE', self._ctx.access_token, self._ctx.refresh_token).data
